configure.ac

```diff
@@ -548,12 +546,41 @@ if test -z "$ENABLE_SOS_TRUE"; then
 	fi
 fi
 
-dnl ldms-python API's
+dnl ldms-python API's and programs
+AX_WITH_PROG([CYTHON], [cython], [no])
+CYTHON_REQ_VER="0.28.5"
+CYTHON_MSG="
+
+    Please install Cython ver >= ${CYTHON_REQ_VER}. It is recommended to install
+    Cython via pip3 (\`pip3 install Cython\`) to get the latest version of
+    Cython.  Some installation package (e.g. python36-Cython from epel) may
+    install cython program with cython<VER> form (e.g. /bin/cython3.6). In such
+    case, --with-cython=FULL_PATH_TO_CYTHON_PROGRAM option is required (e.g.
+    --with-cython=/bin/cython3.6).
+
+"
 if test -z "$ENABLE_LDMS_PYTHON_TRUE"; then
 	AX_PYTHON_DEVEL([>='3.6'])
 	pkgpythondir="${pythondir}/ovis_ldms"
 	pkgpyexecdir="${pkgpythondir}"
 	AX_PYTHON_MODULE_VERSION(["argparse"], 1.1, "python")
+
+	dnl cython check
+	if test "${CYTHON}" == "no"; then
+		dnl cython not found, try cython3 or cython3.6
+		AC_MSG_NOTICE(cython not found ... trying cython3 or cython3.6)
+		CYTHON=$(which cython3 2>/dev/null || \
+			 which cython3.6 2>/dev/null || \
+			 echo "no")
+		test "${CYTHON}" == "no" || AC_MSG_NOTICE(Using cython: ${CYTHON})
+	fi
+	test "${CYTHON}" != "no" || \
+		AC_MSG_ERROR(`cython` program not found. $CYTHON_MSG)
+	CYTHON_VER=$(${CYTHON} --version 2>&1 | sed -e 's/.*\s//')
+	AC_MSG_NOTICE(Cython version: ${CYTHON_VER})
+	LE_VER=$(echo -e "${CYTHON_VER}\n${CYTHON_REQ_VER}"| sort -V|head -n1)
```

Comment:
Please use AX_COMPARE_VERSION for this.

Reply:
Why?

Reply:
Because it is almost always better to use an existing macro than to invent our own sh scripting.

Reply:
I would agree if AX_COMPARE_VERSION is a built-in. It is not though (on my CentOS7 machine). I need to download the macro from http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_compare_version.m4. Comparing to just the two-liner here ... I think I'll keep it as-is.